games-emulation/retroarch/retroarch-1.7.1.recipe

```diff
@@ -0,0 +1,60 @@
+SUMMARY="The reference frontend of libretro"
+DESCRIPTION="RetroArch is a frontend for emulators, game engines and media players. \
+It enables you to run classic games on a wide range of computers and consoles \
+through its slick graphical interface. It has advanced features like shaders, \
+netplay, rewinding, next-frame response times, and more!"
+HOMEPAGE="https://libretro.com"
+COPYRIGHT="2010-2018 The RetroArch Team"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/libretro/RetroArch/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="e9189c7da6b2ca0142528eca14f26104ee322f052f8609203d90b60938d8d3f3"
+SOURCE_DIR="RetroArch-$portVersion"
+PATCHES="retroarch-$portVersion.patchset"
+
+ARCHITECTURES="x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	retroarch$secondaryArchSuffix = $portVersion
+	app:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch$secondaryArchSuffix = $portVersion
+	cmd:retroarch_cg2glsl$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libsdl2_2.0$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	ffmpeg$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsdl2$secondaryArchSuffix
+	devel:libgl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:which
+	cmd:git
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	"
+GLOBAL_WRITABLE_FILES="
+	$relativeSettingsDir/retroarch.cfg keep-old
+"
```

Comment:
Please note that `USER_SETTINGS_FILES` is only used for files that are not included in the package. Consult [BuildingPackages#The .PackageInfo][0] for more details

>      

> `user-settings-files`: A list of user settings file infos. The list elements must have the following format:
>
>       user_settings_file_info     ::= path [ "directory" | "template" template_path ]
>
> `path` is the relative path of the settings file or directory, starting with "settings/". It is not included in the package. However, if `template_path` is specified, it is a path to a file included in the package that can serve as a template for the settings file. It doesn't imply any automatic action on package activation, though. If the "directory" keyword is given, the path refers to a settings directory (typical when a program creates multiple settings files).


[0]: https://github.com/alaviss/haiku/blob/wiki-fixes/docs/develop/packages/BuildingPackages.rst#the-packageinfo

Reply:
I might be wrong, but what I have understood from the recipes I've seen so far is that `USER_SETTINGS_FILES` is used to inform pkgman that a given config in `/system/settings/` has a a read-only template. Here we could use `/system/data/retroarch/retroarch.cfg.default` for that default.
BTW, I did not kick a build yet, so I don't know if `INSTALL()` will install the `retroarch.cfg`. In case it does not, then all we need to do is add a `install -t "$settingsDir" retroarch.cfg` in `INSTALL()`.

If we're not allowed to use `USER_SETTINGS_FILES` for config files that we install, then there are cannot provide templates for them. My undestanding is that the template provides sane defaults that the user can copy in case the config does not work.

Reply:
Hello folks,

To make things clear, `/system/settings/retroarch.cfg` is indeed a read-only skeleton file, used by retroarch upon its first run as a template to generate a user-specific settings file in `$XDG_CONFIG_HOME/retroarch/retroarch.cfg`